pkg/abap/build/connector.go

```diff
@@ -174,3 +176,39 @@ func (conn Connector) UploadSarFile(appendum string, sarFile []byte) error {
 	defer response.Body.Close()
 	return nil
 }
+
+// UploadSarFileInChunks : upload *.sar file in chunks
+func (conn Connector) UploadSarFileInChunks(appendum string, fileName string, sarFile []byte) error {
+	//Maybe Next Refactoring step to read the file in chunks, too?
+
+	url := conn.Baseurl + appendum
+
+	header := make(map[string][]string)
+	header["Content-Disposition"] = []string{"form-data; name=\"file\"; filename=\"" + fileName + "\""}
+
+	//chunkSize := 10000 // 10KB for testing
+	//chunkSize := 1000000 //1MB for Testing,
+	chunkSize := 10000000 //10MB
```

Comment:
This chunk size seems a bit large to me. Does it not increase the chance of failure and also the cost in case you add retries as suggested below?

Reply:
As of now we did not have any reliability issues - idea was to use a value which is not too close to the actual limit but also not too far away to produce too much overhead (I would no like to have hundereds or thousands of iterations)

Reply:
added the idea of retry as comment in the method